src/execute/julia.ts

```diff
@@ -600,16 +600,21 @@ async function writeJuliaCommand(
   // but they could also end in a partial one.
   // so to read and process them all correctly, we read in a fixed number of bytes, if there's a newline, we process
   // the string up to that part and save the rest for the next round.
-  let restOfPreviousResponse = "";
+  let restOfPreviousResponse = new Uint8Array(512);
+  let restLength = 0; // number of valid bytes in restOfPreviousResponse
   while (true) {
-    let response = restOfPreviousResponse;
-    const newlineAt = response.indexOf("\n");
+    const respArray : Uint8Array[] = [];
+    let respLength = 0;
+    let response = "";
+    const newlineAt = restOfPreviousResponse.indexOf(10);
```

Comment:
(Leaving a comment here for someone else who might wonder)

Searching for a naked `10` value in a naked UTF-8 stream looks a bit suspicious, but actually does work. The UTF-8 encoding is clever in that all extended code units appear in the byte stream as bytes that have `10` in [their most significant two bits](https://en.wikipedia.org/wiki/UTF-8#Encoding). Nice!